syntax = "proto3";

// python -m grpc_tools.protoc --proto_path=protobuf --python_out=backend --grpc_python_out=backend protobuf/jobs.proto  

message Job {
  string id = 1;
  string title = 2;
  string company = 3;
  string location = 4;
  int64 salary = 5;
}

message JobList {
  repeated Job jobs = 1;
}

message JobId {
  string id = 1;
}

message Empty {}

// Service 
service JobService {
  // Create
  rpc AddJob (Job) returns (Job) {}
  // Read 
  rpc GetAll (Empty) returns (JobList) {}
  rpc GetJob (JobId) returns (Job) {}
  // Update
  rpc UpdateJob (Job) returns (Job) {}
  // Delete
  rpc DeleteJob (JobId) returns (Empty) {}
}
